src/StringOrArrayDynamicFunctionReturnTypeExtension.php

```diff
@@ -35,7 +35,7 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
         $dataArgType = $scope->getType($dataArg);
         if ($dataArgType->isArray()->yes()) {
             $keyType = $dataArgType->getIterableKeyType();
-            if ($keyType instanceof StringType) {
+            if ($keyType->isString()->yes()) {
                 return new ArrayType(new StringType(), new StringType());
             }
             return new ArrayType(new IntegerType(), new StringType());
```

Comment:
```suggestion
            if ($keyType->isString()->no()) {
                return new ArrayType(new IntegerType(), new StringType());
            }
            return TypeCombinator::union(
                new ArrayType(new StringType(), new StringType()),
                new ArrayType(new IntegerType(), new StringType())
            );
```

Reply:
I see what you're going for I think. I don't really understand though what this extension is even doing :) I was just replacing stuff without changing much behaviour.
Why does it not just pass through the array but recreate it for example. Unfortunately not  on my computer anymore to dig deeper..

Reply:
It sets the value type to string no matter what. However, this is not what the extension was intended to I think. See #154.

Reply:
I'll check it later and try to add tests too it not existing yet. But does it make sense to you to directly pass through the key type? It should noy be modified, right?